### PR TITLE
Fixes #19.

### LINKED ISSUE #19: Support passing auth token as request param

### PR DESCRIPTION
**Pull request by jweakley**
This will require a version bump to 0.5.0 and publish.

### CODE REVIEW

File: lib/token_authenticate_me/controllers/token_authenticateable.rb [hunk 1]
```diff
@@ -34,6 +30,16 @@ def authenticate_token
...
       end
 
+      def authenticate_param
+        token = params[:authentication_token]
+        session = Session.find_by_key(token)
+        @session ||= if session && (session.expiration > DateTime.now)
+                       session
+                     else
+                       false
+                     end
```
Comment (Samsinite) [inline lib/token_authenticate_me/controllers/token_authenticateable.rb]: Some weird spacing going on here...

**Comment by Samsinite** [comment]
:+1: after addressing the last rubocop issue and the spacing.

File: lib/token_authenticate_me/controllers/token_authenticateable.rb (hunk 1 above)
Comment (jweakley) [inline lib/token_authenticate_me/controllers/token_authenticateable.rb]: Rubocop probably did this to keep the if/else vertically alined.

Comment (Samsinite) [inline lib/token_authenticate_me/controllers/token_authenticateable.rb]: Hmm... now that I look at this, I see that there is both instance variables and class instance variables, which is a little confusing. Why are we `@session ||= ...` here?

**Comment by Samsinite** [comment]
FWIW, in my branch, I pulled out the token auth strategy into a method so it is overridable (https://github.com/wildland/token_authenticate_me/blob/restrucure-lib-and-add-invites/lib/token_authenticate_me/concerns/controllers/token_authenticateable.rb#L34). If you take this approach, then `authenticate_token` can become something like:

```rb
  def authenticate_token
    @session ||= (authenticate_with_http_token(&method(:token_handler)) || authenticate_with_params)
  end

  def authenticate_with_params
    token = params[:authentication_token]
    token_handler(token, {})
  end
```

**Comment by Samsinite** [comment]
^^^ would also fix the code duplication that you have going on.

File: lib/token_authenticate_me/controllers/token_authenticateable.rb [hunk 2]
```diff
@@ -12,32 +12,38 @@ module TokenAuthenticateable
       protected
 
       def authenticate
-        authenticate_token || render_unauthorized
+        authenticate_token || authenticate_param || render_unauthorized
```
Comment (Samsinite) [inline lib/token_authenticate_me/controllers/token_authenticateable.rb]: only need:
```rb
authenticate_token || render_unauthorized
```

Now

**Comment by Samsinite** [comment]
:+1:, once tested in denovo or a similar app.

**Pull request merged**